internal/pkg/cli/svc_init.go

```diff
@@ -305,30 +335,42 @@ func (o *initSvcOpts) askSvcName() error {
 }
 
 // askDockerfile prompts for the Dockerfile by looking at sub-directories with a Dockerfile.
-// If the user chooses to enter a custom path, then we prompt them for the path.
 func (o *initSvcOpts) askDockerfile() error {
 	if o.DockerfilePath != "" {
 		return nil
 	}
 
-	// TODO https://github.com/aws/copilot-cli/issues/206
 	dockerfiles, err := listDockerfiles(o.fs, ".")
-	if err != nil {
-		return err
+	// If Dockerfiles are found in the current directory or subdirectory one level down, ask the user to select one.
+	var sel string
+	if err == nil {
+		sel, err = o.prompt.SelectOne(
+			fmt.Sprintf(fmtSvcInitDockerfilePrompt, color.Emphasize("Dockerfile"), color.HighlightUserInput(o.Name)),
+			svcInitDockerfileHelpPrompt,
+			dockerfiles,
+			prompt.WithFinalMessage("Dockerfile:"),
+		)
+		if err != nil {
+			return fmt.Errorf("select Dockerfile: %w", err)
+		}
+		o.DockerfilePath = sel
+		return nil
 	}
 
-	sel, err := o.prompt.SelectOne(
-		fmt.Sprintf(fmtSvcInitDockerfilePrompt, color.Emphasize("Dockerfile"), color.HighlightUserInput(o.Name)),
-		svcInitDockerfileHelpPrompt,
-		dockerfiles,
-		prompt.WithFinalMessage("Dockerfile:"),
-	)
+	// If no Dockerfiles were found, prompt user for custom path.
+	var notExistErr *dockerfile.ErrDockerfileNotFound
+	if !errors.As(err, &notExistErr) {
+		return err
+	}
+	sel, err = o.prompt.Get(
+		fmt.Sprintf(fmtSvcInitDockerfilePathPrompt, color.Emphasize("Dockerfile"), color.HighlightUserInput(o.Name)),
+		svcInitDockerfilePathHelpPrompt,
+		validatePath,
+		prompt.WithFinalMessage("Path to Dockerfile:"))
```

Comment:
nit: for consistency
```suggestion
		prompt.WithFinalMessage("Dockerfile:"))
```